spacy_llm/tasks/ner/task.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Callable, Dict, Iterable, List, Optional, Type
+
+from spacy.language import Language
+from spacy.scorer import get_ner_prf
+from spacy.tokens import Doc, Span
+from spacy.training import Example
+from spacy.util import filter_spans
+
+from ...compat import Literal
+from ...ty import TaskResponseParserProtocol
+from ..span import SpanExample, SpanTask
+from ..templates import read_template
+
+DEFAULT_NER_TEMPLATE_V1 = read_template("ner.v1")
+DEFAULT_NER_TEMPLATE_V2 = read_template("ner.v2")
+
+
+class NERTask(SpanTask):
+    def __init__(
+        self,
+        labels: List[str],
+        template: str,
+        parse_responses: TaskResponseParserProtocol,
+        fewshot_example_type: Type[SpanExample],
+        label_definitions: Optional[Dict[str, str]],
+        prompt_examples: Optional[List[SpanExample]],
+        normalizer: Optional[Callable[[str], str]],
+        alignment_mode: Literal["strict", "contract", "expand"],
+        case_sensitive_matching: bool,
+        single_match: bool,
+    ):
+        """Default NER task.
+
+        labels (List[str]): List of labels to pass to the template.
+            Leave empty to populate it at initialization time (only if examples are provided).
+        template (str): Prompt template passed to the model.
+        parse_responses (TaskResponseParser): Callable for parsing LLM responses for this task.
+        fewshot_example_type (Type[FewshotExample]): Type to use for fewshot examples.
+        label_definitions (Optional[Dict[str, str]]): Map of label -> description
+            of the label to help the language model output the entities wanted.
+            It is usually easier to provide these definitions rather than
+            full examples, although both can be provided.
+        examples (Optional[Callable[[], Iterable[Any]]]): Optional callable that
```

Comment:
Do we want to call this attribute `prompt_examples` or `fewshot_examples` or `examples`? This is a inconsistent currently.

Reply:
Definitely not `examples`. I thought the current name was `prompt_examples`? Though I'm happy with `fewshot_examples`, too.

Reply:
Some were `prompt_examples`, some were `examples`. I'll go with `fewshot_examples` then, I think it's the most recognizable term.

Reply:
Done in https://github.com/explosion/spacy-llm/pull/242/commits/b443670e1a7c407fcb637e32b9088b8b9b5045b7. However, changing `examples` to `fewshot_examples` also in the factory functions (as it's currently the case in this PR) is a breaking change. Bumping task versions just for this also seems like a bother. 

Should we revert `fewshot_examples` to `examples` in the task factory functions?

Reply:
As discussed internally, let's keep the registered task definitions as-is to avoid too many breaking changes in existing configs, and to minimize the refactoring done in this PR, which is difficult to track diff-wise.